desktop/flipper-server-core/src/devices/android/adbClient.tsx

```diff
@@ -37,16 +36,24 @@ export async function initializeAdbClient(
    however, it sometimes fails with ENOENT errors. So instead, we start it
    manually before requesting a client. */
 async function createClient(config: Config): Promise<Client> {
-  const androidHome = config.androidHome;
-  let adbPath = path.resolve(androidHome, 'platform-tools', 'adb');
-  if (!(await pathExists(adbPath))) {
-    console.info('falling back to the alternative adb path');
-    adbPath = path.resolve(androidHome, 'adb');
-  }
   return reportPlatformFailures<Client>(
-    execFile(adbPath, ['start-server']).then(() =>
+    startAdbServer(config.androidHome).then(() =>
       adbkit.createClient(adbConfig()),
     ),
     'createADBClient.shell',
   );
 }
+
+async function startAdbServer(androidHome: string) {
+  const adbPath = path.resolve(androidHome, 'platform-tools', 'adb');
+  const args = ['start-server'];
+
+  return execFile(adbPath, args).catch((error) => {
+    if (error.code == 'ENOENT') {
```

Comment:
The errors returned by child process are standard 
There's nothing mentioned here that the code can be different depending on platform https://nodejs.org/api/errors.html#common-system-errors

We can fallback for any error, but this will hide the original error 
So if the original error is **not** "file not found" using the fallback path would not help, but mask the root error